lib/awful/ewmh.lua

```diff
@@ -373,7 +373,7 @@ end
 -- @tparam string context The context
 -- @tparam[opt={}] table hints The hints to pass to the handler
 function ewmh.client_geometry_requests(c, context, hints)
-    if context == "ewmh" and hints then
+    if context == "ewmh" and hints and not c.immobilized then
```

Comment:
Huh? What happened to the code that checked the individual properties? Something like
```lua
local geo = {}
if not c.immobilized_horizontal then
    geo.x = hints.x
    geo.width = hints.width
end
if not c.immobilized_vertical then
    geo.y = hints.y
    geo.height = hints.height
end
c:geometry(geo)
```